### ISSUE TITLE
Support for ESP32

### ISSUE THREAD

**Issue by antonio-fiol** [ThingPulse/esp8266-weather-station#71]
I bought one of these little creatures... https://es.aliexpress.com/store/product/Lolin-ESP32-OLED-wemos-for-Arduino-ESP32-OLED-WiFi-Modules-Bluetooth-Dual-ESP-32-ESP-32S/1983387_32807531243.html
And I thought it would be cool to run your code there.  Why not?
Well, I found a few compilation issues, and then a very weird behavior when accessing the APIs, because available() returned 0 after checking if it was still connected. Bug or ESP32 feature? I don't know.

Please find a patch attached that "works for me".
[esp32-support-draft.patch.txt](https://github.com/squix78/esp8266-weather-station/files/1082679/esp32-support-draft.patch.txt)

Please note that the examples are NOT included in the patch, but they might need changing the #include for the WiFi.

For the specific device, the OLED  display pins need to be adjusted according to their product description.

Hoping that posting this here helps someone.

**Comment by squix78** [comment]
Cool! Thank you so much. I hope I will find soon to test it myself!

**Comment by antonio-fiol** [comment]
Please find attached a stripped version of the ExtendedDST demo.
[ESP32_WeatherStationDemoExtendedDST.zip](https://github.com/squix78/esp8266-weather-station/files/1095632/ESP32_WeatherStationDemoExtendedDST.zip)

This is the one I modified because I wanted DHT11 support as well.

Summary of changes:

- removed WifiManager (requires ESP8266WiFi).
- removed mDNS, DNSServer and WebServer imports (not used [?] and require ESP8266)
- removed OTA (code to determine the hostname that is passed to OTA needs ESP.getChipId, which is 8266-specific). Could have used something else, but I did not need OTA anyway.
- removed Ticker (it is only available for ESP8266), created substitution code for the Ticker bit.

In your settings.h, you'll need to add the WIFI connection information constants for this sketch to compile.

I powered the DHT11 from the 3V3 output. I tried to power it from a GPIO pin, in order to disable it while not reading and save a few mA, but that did not work for me.

**Comment by antonio-fiol** [comment]
For some reason, the clock on the demo I uploaded on my last comment seems off. I didn't test much, but it seems that passing configTime an offset with the opposite sign would fix it.
`configTime(-(UTC_OFFSET * 3600), 0, NTP_SERVERS)`
But this looks terribly wrong. Any idea why configTime would behave "backwards"?

**Comment by bill-orange** [comment]
I was able to get the WeatherStationDemo and OpenWeatherMapForecastDemo using OpenWeatherMap running on an ESP32 without too much trouble.  Here's what I discovered.

1. On the Wemos-TTGO-ESP32 boards (which are usually counterfit!) the I2C pin connections are okay as-is.

2.  You will need to comment out the line `#include <coredecls.h>`  if present.

3. The compiler will will throw out errors on a couple of integer types.  Note in which file the error is found and add: 

```
typedef signed short        sint16_t;
typedef signed long         sint32_t;
```
Do this as needed.

That's it.  I changed my time zone and daylight savings time info and it ran fine.

I suggest that on the next update to the library the typedef statements should be added to the library files.  This will streamline ESP32 use and should not effect ESP 8266 use.  The coredecl.h include could be inside an ifdef.
